pkg/phases/postgresoperator/test.go

```diff
@@ -45,11 +49,148 @@ func Test(p *platform.Platform, test *console.TestResults) {
+		return
+	}
+	backupPaths, err := db1.ListBackups("", 1, true)
+	if err != nil {
+		test.Failf("failed to get list of backups for Postgresql Cluster %s: %v", cluster1ZalandoPsqlName, err)
+		return
+	}
+	if len(backupPaths) < 1 {
+		test.Failf("there is no backup found for Postgresql Cluster %s. Expected at least 1.", cluster1ZalandoPsqlName)
+		return
+	}
+	test.Passf(testName, "get the latest backup of %s", cluster1ZalandoPsqlName)
+	db2, err := pgclient.GetPostgresDB(&p.Client, cluster2ZalandoPsqlName)
+	if err != nil {
+		test.Failf("failed to get Postgres Client for %s: %v", cluster2ZalandoPsqlName, err)
+		return
+	}
+	backupPathToRestore := backupPaths[0]
+	err = db2.Restore(backupPathToRestore)
+	if err != nil {
+		test.Failf("failed to restore backup %s to Postgresql Cluster %s: %v", backupPathToRestore, cluster2ZalandoPsqlName, err)
+		return
+	}
+	test.Passf(testName, "restore logical backup \"%s\" of %s to %s successfully", backupPathToRestore, cluster1ZalandoPsqlName, cluster2ZalandoPsqlName)
+	if err := testFixturesArePresent(p, cluster2ZalandoPsqlName, 3*time.Minute, test); err != nil {
+		test.Failf(testName, "failed to find test fixtures data in PG Cluster %s: %v", cluster2ZalandoPsqlName, err)
+		return
+	}
+	test.Passf(testName, "verified test fixtures is in %s after the restore", cluster2ZalandoPsqlName)
+}
+
+func newPostgresqlDBCluster(namespace, name string) *postgresdbv1.PostgresqlDB {
+	return &postgresdbv1.PostgresqlDB{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "PostgresqlDB",
+			APIVersion: "db.flanksource.com/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("e2e-test-%s-%s", utils.RandomString(6), name),
+			Namespace: namespace,
+		},
+		Spec: postgresdbv1.PostgresqlDBSpec{
+			Storage: postgresdbv1.Storage{
+				Size: "500Mi",
+			},
+			Backup: postgresdbv1.PostgresqlBackup{
+				Bucket: fmt.Sprintf("logical-backup-test-%s", utils.RandomString(6)),
+				Retention: postgresdbv1.BackupRetention{
+					KeepLast: 5,
+				},
+			},
+			Replicas: 1,
+		},
+		Status: postgresdbv1.PostgresqlDBStatus{
+			Conditions: []v1.Condition{},
+		},
+	}
+}
+
+func removeLogicalBackupE2ECluster(p *platform.Platform, test *console.TestResults, db *postgresdbv1.PostgresqlDB) {
+	client, _, _, err := p.GetDynamicClientFor(Namespace, db)
+	if err != nil {
+		test.Errorf("Failed to get dynamic client: %v", err)
+		return
+	}
+
+	_, err = client.Get(context.TODO(), db.Name, metav1.GetOptions{})
+	if kerrors.IsNotFound(err) {
+		return
+	}
+
+	if err := client.Delete(context.TODO(), db.Name, metav1.DeleteOptions{}); err != nil {
+		test.Warnf("Failed to delete resource %s/%s/%s in namespace %s", db.APIVersion, db.Kind, db.Name, Namespace)
+		return
 	}
+	test.Infof("Deleted %s/%s/%s in namespace %s", db.APIVersion, db.Kind, db.Name, Namespace)
 }
 
-func TestE2E(p *platform.Platform, test *console.TestResults) {
+func TestCloneDBFromWAL(p *platform.Platform, test *console.TestResults) {
```

Comment:
I don't know why this test failed for me locally (using Kind) but passed in CICD. I haven't had time to dig into that yet.

@moshloop @teodor-pripoae